src/js/mqtt.js

```diff
@@ -283,12 +288,16 @@ MqttClient.prototype.disconnect = function(err) {
  */
 MqttClient.prototype.publish = function(topic, payload, options, callback) {
   callback = callback || noop;
+  if (typeof options === 'function') {
+    callback = options;
+  }
+
   if (!Buffer.isBuffer(payload)) {
     payload = new Buffer(payload);
   }
   try {
     var buf = this._handle._getPublish(topic, {
-      id: this._msgId++,
+      id: qosEqual0(options && options.qos) ? 0 : ++this._msgId,
```

Comment:
++ is not good way to represent, the root reason is that the `msgId` is starting from zero, could that start from 1 to fix this problem?

Reply:
But should it be the different value when different publish?

Reply:
Yeah, may ++ is not very readable:)